### PR TITLE
(PE-36454) add support for adding attributes to csrs, fix issues

### PR DESCRIPTION
**Pull request by jonathannewman**
This adds support in the "simple" namespace for adding attributes to CSRs. It adds a test to demonstrate the behavior.

It also fixes several problems, including a typo in the "get-keys" function which was associating a name with the wrong key, as well as removing use of a deprecated conversion routine.

Some require organization was done as well.

### CODE REVIEW

**Comment by jonathannewman** [comment]
This repo doesn't currently have automated testing.

Local testing showed:
```
lein test puppetlabs.ssl-utils.core-test

lein test puppetlabs.ssl-utils.extensions-utils-test

lein test puppetlabs.ssl-utils.simple-test

Ran 28 tests containing 261 assertions.
0 failures, 0 errors.
```

File: src/clojure/puppetlabs/ssl_utils/simple.clj [hunk 1]
```diff
@@ -117,7 +117,7 @@
    (let [validity (cert-validity-dates (* 5 60 60 24 365))
          extensions (if ca?
                       (ssl-utils/create-ca-extensions
-                        (:x509-name host-keys)
+                        (:x500-name host-keys)
```
Comment (jonathannewman) [inline src/clojure/puppetlabs/ssl_utils/simple.clj:120]: This was a typo and was causing test failures.

**Comment by steveax** [comment]
tests passed for me locally too

**Review by steveax — APPROVED** [review]
👍

**Pull request merged**